src/components/DevToolsUi.tsx

```diff
@@ -5,80 +5,123 @@ import { Tabs, TabsContent, TabsList, TabsTrigger } from '@/components/ui/tabs'
 import { DiagnosticData } from '@/types/DiagnosticData'
 
 import { ConsoleOutput } from './ConsoleOutput'
-import { CreateGithubIssue } from './CreateGithubIssue'
+import { DescriptionInfo } from './DescriptionInfo'
+import { Header } from './Header'
 import { LogsTable } from './LogsTable'
-import { MetadataTable } from './MetadataTable'
-import { QueryResults } from './QueryResults'
+import { QueryResultsPanel } from './QueryResultsPanel'
 import { RawContent } from './RawContent'
 
 interface DevToolsUIProps {
   diagnosticData: DiagnosticData
   slackFileId?: string
 }
 
+interface TabHeaderProps {
+  id: string
+  title: string
+  count: number | null
+  variant?: 'destructive' | 'light'
+}
+
+const TabHeader = ({ id, title, count, variant = 'light' }: TabHeaderProps) => {
+  if (count == null) {
+    return null
+  }
```

Comment:
moving this conditional in here lets us conditionally render these while making the parent component much more readable

Reply:
Indeed!